include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -391,57 +404,70 @@ class pairwise_alignment_edit_distance_unbanded
         _compute();
         result_value_type res_vt{};
         res_vt.id = idx;
-        if constexpr (!std::is_same_v<decltype(res_vt.score), std::nullopt_t *>)
+        if constexpr (compute_score)
         {
             res_vt.score = score();
         }
 
-        if constexpr (!std::is_same_v<decltype(res_vt.back_coordinate), std::nullopt_t *>)
+        if constexpr (compute_back_coordinate)
         {
             res_vt.back_coordinate = back_coordinate();
         }
 
-        [[maybe_unused]] alignment_trace_matrix matrix = trace_matrix();
-        if constexpr (!std::is_same_v<decltype(res_vt.front_coordinate), std::nullopt_t *>)
+        if constexpr (compute_front_coordinate)
         {
-            res_vt.front_coordinate = alignment_front_coordinate(matrix, res_vt.back_coordinate);
+            res_vt.front_coordinate = alignment_front_coordinate(trace_matrix(), res_vt.back_coordinate);
         }
 
-        if constexpr (!std::is_same_v<decltype(res_vt.alignment), std::nullopt_t *>)
+        if constexpr (compute_sequence_alignment)
         {
-            res_vt.alignment = alignment_trace(database, query, matrix, res_vt.back_coordinate);
+            res_vt.alignment = alignment_trace(database, query, trace_matrix(), res_vt.back_coordinate);
         }
         return alignment_result<result_value_type>{std::move(res_vt)};
     }
 
     //!\brief Return the score of the alignment.
     score_type score() const noexcept
     {
+        static_assert(compute_score, "score() can only be computed if you specify the result type within your alignment "
+                                     "config, e.g. `align_cfg::edit | align_cfg::result{with_score}`");
```

Comment:
Also it is helpful to give an example in the error message, but I would not do it. In case we change something before 3.1 in the API we might not detect this here.

Reply:
I could add a decltype here to check the syntax xD

Reply:
I removed the .e.g. part